### PR TITLE
fix order of non reservable notification error text checking

### PR DESCRIPTION
**Pull request by vincit-matu**
## 🛠️ Changelog

- Corrects the logic behind what is shown in the notification for a reservation unit not being reservable
- Splits said notification to its own component `NonReservableNotification`, for clearer code

## 🧪 Test plan

- Check that the code seems reasonable.

## 🎫 Tickets

- TILA-3075

### CODE REVIEW

File: apps/ui/components/reservation-unit/Head.tsx [hunk 1]
```diff
@@ -107,6 +107,38 @@ const UnitName = styled(H3).attrs({ as: "h2" })`
...
+              date: reservationUnit.reservationBegins
+                ? formatDate(reservationUnit.reservationBegins, "d.M.yyyy")
+                : "",
+              time: reservationUnit.reservationBegins
+                ? formatDate(reservationUnit.reservationBegins, "H.mm")
+                : "",
+            })
+          : "notReservable"
+    }`
+  );
```
Comment (joonatank) [inline apps/ui/components/reservation-unit/Head.tsx:132]: This is too hard to follow (nested ternaries). I'd separate the error message generation to a function, but anything other than multiple ternaries (define the messages as variables first, then pick the right one in one select etc.).

**Review by joonatank — COMMENTED** [review]
It's an improvement, but I can't read the error message generation logic.

File: apps/ui/components/reservation-unit/Head.tsx [hunk 2]
```diff
@@ -107,6 +107,40 @@ const UnitName = styled(H3).attrs({ as: "h2" })`
...
+            ? formatDate(reservationUnit.reservationBegins, "H.mm")
+            : "",
+        })
+      : false;
+  const returnText = t(
+    `reservationUnit:notifications.${
+      reservationUnit.reservationKind ===
+      ReservationUnitsReservationUnitReservationKindChoices.Season
+        ? "onlyRecurring"
+        : getFutureText() || "notReservable"
```
Comment (joonatank) [inline apps/ui/components/reservation-unit/Head.tsx:132]: Is the future text supposed to be inside the translation?

Comment (vincit-matu) [reply apps/ui/components/reservation-unit/Head.tsx]: Nope. Refactored the whole thing to a more readable form.

**Review by joonatank — APPROVED** [review]
👍
